examples/ron/src/main.rs

```diff
@@ -0,0 +1,83 @@
+#![feature(proc_macro_hygiene)]
+
+#[macro_use] extern crate rocket;
+#[macro_use] extern crate serde_derive;
+
+#[cfg(test)] mod tests;
+
+use std::sync::Mutex;
+use std::collections::HashMap;
+
+use rocket::State;
+use rocket_contrib::ron::{Ron};
+
+// The type to represent the ID of a message.
+type ID = usize;
+
+// We're going to store all of the messages here. No need for a DB.
+type MessageMap = Mutex<HashMap<ID, String>>;
+
+#[derive(Serialize, Deserialize)]
+struct Message {
+    id: Option<ID>,
+    contents: String
+}
+
+#[derive(Serialize, Deserialize)]
```

Comment:
This can be only `derive(Serialize)`, since it's never used as input to the server.

Reply:
Oops, didn't realize that.